Aggregate uploader: Gradle migration and Java9 compat

ggalmazor opened the pull request:
This PR migrates the project to Gradle and configures the shadow plugin to produce an uberJAR that will run under both Java8 and Java9

Tagging @lognaturel and @yanokwa to avoid missing this PR :)

yanokwa commented:
Alrighty, I've figured out `./gradlew shadowJar` works. The code is good, but I'd love some changes to the readme to make this process more obvious.

ggalmazor commented:
Since this repo is really a bunch of projects and only this one seems to be really maintained, I've added a README file to the project subdirectory. 

I've used Aggregate's readme file as a template and I've added a small section explaining how to build the JAR with dependencies.

I'll add a note explaining the relation between this repo/project and Aggregate in the Aggregate repo.
